src/battle_script_commands.c

```diff
@@ -1171,13 +1171,15 @@ bool32 IsBattlerProtected(u8 battlerId, u16 move)
         return FALSE;
     else if (gBattleMoves[move].effect == MOVE_EFFECT_FEINT)
         return FALSE;
-    else if (gProtectStructs[battlerId].protected)
+    else if (gProtectStructs[battlerId].protected && move != MOVE_DECORATE)
         return TRUE;
     else if (gSideStatuses[GetBattlerSide(battlerId)] & SIDE_STATUS_WIDE_GUARD
              && gBattleMoves[move].target & (MOVE_TARGET_BOTH | MOVE_TARGET_FOES_AND_ALLY))
         return TRUE;
     else if (gProtectStructs[battlerId].banefulBunkered)
         return TRUE;
+    else if (gProtectStructs[battlerId].obstruct && !IS_MOVE_STATUS(move))
+        return TRUE;
     else if (gProtectStructs[battlerId].spikyShielded)
         return TRUE;
     else if (gProtectStructs[battlerId].kingsShielded && gBattleMoves[move].power != 0)
```

Comment:
I wonder if some of these `gBattleMoves[move].power != 0` checks should be changed to `!IS_MOVE_STATUS(move)` checks since I think they're meant to represent a check for if it's a status move being used.